optimlol_api/controllers/summonerController.js

```diff
@@ -148,7 +148,12 @@ module.exports = function() {
 				var championStats = results[promiseObject.STATS_INDEX].state === 'fulfilled' ? results[promiseObject.STATS_INDEX].value : null;
 				var recentHistoryStats = results[promiseObject.RECENT_STATS_INDEX].state === 'fulfilled' ? results[promiseObject.RECENT_STATS_INDEX].value : null;
 				var champions = results[promiseObject.CHAMPIONS_INDEX].state === 'fulfilled' ? results[promiseObject.CHAMPIONS_INDEX].value : null;
-				
+
+				summoner.dataQuality = "fresh";
+				if(championStats.quality === "stale" || recentHistoryStats.quality === "stale" || champions.quality === "stale") {
+					summoner.dataQuality = "stale";
+				}
+
```

Comment:
On line 43 of the summonerController we only pass the data object.

_getStats only returns the data object.